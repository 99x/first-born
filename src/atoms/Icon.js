import { Platform } from "react-native";
import React, { Component } from "react";
import Ionicon from "react-native-vector-icons/Ionicons";
import PropTypes from "prop-types";
import { commonColors } from "../utils/color";

export class Icon extends Component {
    render() {
        const { name, size, color, ...otherProps } = this.props;

        if (!name.includes("logo")) {
            return (
                <Ionicon
                    name={
                        Platform.OS === "android" ? "md-" + name : "ios-" + name
                    }
                    size={size}
                    color={color}
                    {...otherProps}
                />
            );
        }
        return (
            <Ionicon name={name} size={size} color={color} {...otherProps} />
        );
    }
}

Icon.propTypes = {
    ...Ionicon.propTypes,
    name: PropTypes.string
};

Icon.defaultProps = {
    color: commonColors.white,
    size: 18
};
