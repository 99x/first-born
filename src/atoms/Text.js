import React, { Component } from "react";
import { Text } from "react-native";
import PropTypes from 'prop-types';
import { commonColors } from "../utils/color";
import { getFontSize } from "../variables/textSizeVariables";

class TextAE extends Component {

    render() {
        const { color, size, bold, children, ...otherProps } = this.props;

        const textColor = !color ? commonColors.black : color;

        const textSize = !size ? "p" : size;

        const textBold = !bold ? "normal" : "bold";

        const textStyle = { color: textColor, fontSize: getFontSize(textSize), fontWeight: textBold };

        return (
            <Text
                ref={c => (this._root = c)}
                style={textStyle}
                {...otherProps}
            >
                {children}
            </Text>
        )
    }
}

TextAE.propTypes = {
    bold: PropTypes.bool,
    color: PropTypes.string,
    size: PropTypes.oneOf(["h1", "h2", "h3", "h4", "h5", "h6", "p", "callout", "sub_heading", "footnote", "caption_big", "caption_small"]),
    ...Text.propTypes
};

TextAE.defaultProps = {
    size: "p",
    bold: false,
    color: commonColors.black
};

export { TextAE as Text };
