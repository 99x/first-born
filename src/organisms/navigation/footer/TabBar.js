import React, { Component } from "react";
import { View, StyleSheet, Platform } from "react-native";
import PropTypes from "prop-types";
import { commonColors } from "../../../utils/color";
import { isIphoneX } from "../../../utils/platform";

export class TabBar extends Component {
    render() {
        const { color, children, ...otherProps } = this.props;

        let tabBarStyle = [styles.container, { backgroundColor: color }];

        if (isIphoneX()) {
            tabBarStyle.push({ paddingBottom: 15 });
        }

        return (
            <View style={tabBarStyle} {...otherProps}>
                {children && children}
            </View>
        );
    }
}

TabBar.propTypes = {
    color: PropTypes.string,
    ...View.propTypes
};

TabBar.defaultProps = {
    color: Platform.OS === "android" ? commonColors.primary : "#f8f8f8"
};

const styles = StyleSheet.create({
    container: {
        flexDirection: "row",
        ...Platform.select({
            ios: {
                borderTopColor: commonColors.inputGrey,
                borderTopWidth: 1
            }
        })
    }
});
