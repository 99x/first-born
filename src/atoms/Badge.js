import React, { Component } from "react";
import { View, StyleSheet, Platform, Image } from "react-native";
import PropTypes from "prop-types";
import { Icon } from "./Icon";
import { commonColors } from "../utils/color";
import { Text } from "./Text";

export class Badge extends Component {
    render() {
        const {
            color,
            children,
            ...otherProps
        } = this.props;

        const badgeColor = color;

        let badgeStyle = [styles.defaultBadge, {
            borderRadius: 15,
            backgroundColor: badgeColor
        }];

        let textStyle = {
            color: commonColors.white
        };

        let iconColor = commonColors.white;;

        const newChildren = React.Children.map(children, child =>
            child && child.type === Text
                ? React.cloneElement(child, {
                    ...child.props,
                    style: { ...child.props.style, ...textStyle },
                    size: "footnote"
                })
                : child && child.type === Icon
                    ? React.cloneElement(child, {
                        ...child.props,
                        color: iconColor,
                        size: 15
                    })
                    : child && child.type === Image
                        ? React.cloneElement(child, {
                            ...child.props,
                            style: {
                                ...child.props.style,
                                width: 15,
                                height: 15
                            }
                        })
                        : null
        );

        return (
            <View
                ref={c => (this._root = c)}
                style={badgeStyle}
                {...otherProps}
            >
                {newChildren}
            </View>
        );
    }
}

Badge.propTypes = {
    color: PropTypes.string,
    ...View.propTypes
};

Badge.defaultProps = {
    color: commonColors.primary,
};

const styles = StyleSheet.create({
    defaultBadge: {
        margin: 5,
        flexDirection: "row",
        justifyContent: "center",
        alignItems: "center",
        paddingHorizontal: 10,
        paddingVertical: 5
    },
});
