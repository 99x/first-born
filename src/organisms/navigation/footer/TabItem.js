import React, { Component } from "react";
import { Platform, StyleSheet, TouchableOpacity, Image } from "react-native";
import PropTypes from "prop-types";

import { Text } from "../../../atoms/Text";
import { commonColors } from "../../../utils/color";
import { Icon } from "../../../atoms/Icon";

export class TabItem extends Component {
    render() {
        const {
            label,
            children,
            activeColor,
            inactiveColor,
            active,
            top,
            ...otherProps
        } = this.props;

        let newChildren = [];

        let topPosition = top && Platform.OS === "android";

        const color = active ? activeColor : inactiveColor;

        let containerStyle = topPosition ? [styles.container, styles.containerTop] : styles.container;

        if (children) {
            newChildren = React.Children.map(children, child =>
                child && child.type === Text
                    ? React.cloneElement(child, {
                        ...child.props,
                        size: topPosition ? "footnote" : "caption_big",
                        color,
                        style: {
                            ...child.props.style,
                            opacity: active ? 1 : 0.8
                        }
                    })
                    : child && child.type === Icon
                        ? React.cloneElement(child, {
                            ...child.props,
                            size: topPosition ? 20 : 28,
                            color,
                            style: {
                                ...child.props.style,
                                opacity: active ? 1 : 0.8,
                                marginRight: topPosition ? 5 : 0
                            }
                        })
                        : child && child.type === Image
                            ? React.cloneElement(child, {
                                style: {
                                    ...child.props.style,
                                    width: topPosition ? 20 : 28,
                                    height: topPosition ? 20 : 28,
                                    opacity: active ? 1 : 0.8,
                                    marginRight: topPosition ? 5 : 0
                                },
                                ...child.props
                            })
                            : null
            );
        }

        return (
            <TouchableOpacity style={containerStyle} {...otherProps}>
                {newChildren && newChildren}
            </TouchableOpacity>
        );
    }
}

TabItem.propTypes = {
    activeColor: PropTypes.string,
    inactiveColor: PropTypes.string,
    active: PropTypes.bool,
    top: PropTypes.bool,
    ...TouchableOpacity.propTypes
};

TabItem.defaultProps = {
    activeColor: Platform.OS === "android" ? commonColors.white : "#0a60ff",
    inactiveColor:
        Platform.OS === "android" ? "rgba(209, 216, 224, 0.8)" : "#8e8e93",
    active: false,
    top: false
};

const styles = StyleSheet.create({
    container: {
        flex: 1,
        alignItems: "center",
        justifyContent: "center",
        paddingVertical: 10
    },
    containerTop: {
        flexDirection: "row"
    }
});
