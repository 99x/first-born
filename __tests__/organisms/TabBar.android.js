import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import { Text } from "../../src/atoms/Text";
import { Icon } from "../../src/atoms/Icon";
import { TabBar } from "../../src/organisms/navigation/footer/TabBar";
import { TabItem } from "../../src/organisms/navigation/footer/TabItem";

jest.mock("Platform", () => {
    const Platform = require.requireActual("Platform");
    Platform.OS = "android";
    return Platform;
});

it("renders TabBar with tab titles", () => {
    const tree = renderer
        .create(
            <TabBar>
                <TabItem active>
                    <Text>ONE</Text>
                </TabItem>
                <TabItem>
                    <Text>TWO</Text>
                </TabItem>
                <TabItem>
                    <Text>THREE</Text>
                </TabItem>
                <TabItem>
                    <Text>FOUR</Text>
                </TabItem>
            </TabBar>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it("renders TabBar with icons", () => {
    const tree = renderer
        .create(
            <TabBar>
                <TabItem active>
                    <Icon name="heart" />
                </TabItem>
                <TabItem>
                    <Icon name="star" />
                </TabItem>
                <TabItem>
                    <Icon name="square" />
                </TabItem>
                <TabItem>
                    <Icon name="egg" />
                </TabItem>
            </TabBar>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it("renders TabBar with tab titles and icons", () => {
    const tree = renderer
        .create(
            <TabBar>
                <TabItem active>
                    <Icon name="heart" />
                    <Text>ONE</Text>
                </TabItem>
                <TabItem>
                    <Icon name="star" />
                    <Text>TWO</Text>
                </TabItem>
                <TabItem>
                    <Icon name="square" />
                    <Text>THREE</Text>
                </TabItem>
                <TabItem>
                    <Icon name="egg" />
                    <Text>FOUR</Text>
                </TabItem>
            </TabBar>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
