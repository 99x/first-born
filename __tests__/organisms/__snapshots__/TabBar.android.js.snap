// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders bottom TabBar with icons 1`] = `
<View
  style={
    Array [
      Object {
        "borderTopColor": "rgba(33, 33, 33, 0.5)",
        "borderTopWidth": 1,
        "flexDirection": "row",
      },
      Object {
        "backgroundColor": "#486c86",
      },
    ]
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#FFF",
            "fontSize": 28,
          },
          Object {
            "marginRight": 0,
            "opacity": 1,
          },
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "rgba(209, 216, 224, 0.8)",
            "fontSize": 28,
          },
          Object {
            "marginRight": 0,
            "opacity": 0.8,
          },
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "rgba(209, 216, 224, 0.8)",
            "fontSize": 28,
          },
          Object {
            "marginRight": 0,
            "opacity": 0.8,
          },
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "rgba(209, 216, 224, 0.8)",
            "fontSize": 28,
          },
          Object {
            "marginRight": 0,
            "opacity": 0.8,
          },
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`renders bottom TabBar with tab titles 1`] = `
<View
  style={
    Array [
      Object {
        "borderTopColor": "rgba(33, 33, 33, 0.5)",
        "borderTopWidth": 1,
        "flexDirection": "row",
      },
      Object {
        "backgroundColor": "#486c86",
      },
    ]
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFF",
            "fontSize": 12,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "opacity": 1,
          },
        ]
      }
    >
      ONE
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "rgba(209, 216, 224, 0.8)",
            "fontSize": 12,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "opacity": 0.8,
          },
        ]
      }
    >
      TWO
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "rgba(209, 216, 224, 0.8)",
            "fontSize": 12,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "opacity": 0.8,
          },
        ]
      }
    >
      THREE
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "rgba(209, 216, 224, 0.8)",
            "fontSize": 12,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "opacity": 0.8,
          },
        ]
      }
    >
      FOUR
    </Text>
  </View>
</View>
`;

exports[`renders bottom TabBar with tab titles and icons 1`] = `
<View
  style={
    Array [
      Object {
        "borderTopColor": "rgba(33, 33, 33, 0.5)",
        "borderTopWidth": 1,
        "flexDirection": "row",
      },
      Object {
        "backgroundColor": "#486c86",
      },
    ]
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#FFF",
            "fontSize": 28,
          },
          Object {
            "marginRight": 0,
            "opacity": 1,
          },
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#FFF",
            "fontSize": 12,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "opacity": 1,
          },
        ]
      }
    >
      ONE
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "rgba(209, 216, 224, 0.8)",
            "fontSize": 28,
          },
          Object {
            "marginRight": 0,
            "opacity": 0.8,
          },
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "rgba(209, 216, 224, 0.8)",
            "fontSize": 12,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "opacity": 0.8,
          },
        ]
      }
    >
      TWO
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "rgba(209, 216, 224, 0.8)",
            "fontSize": 28,
          },
          Object {
            "marginRight": 0,
            "opacity": 0.8,
          },
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "rgba(209, 216, 224, 0.8)",
            "fontSize": 12,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "opacity": 0.8,
          },
        ]
      }
    >
      THREE
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "rgba(209, 216, 224, 0.8)",
            "fontSize": 28,
          },
          Object {
            "marginRight": 0,
            "opacity": 0.8,
          },
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "rgba(209, 216, 224, 0.8)",
            "fontSize": 12,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "opacity": 0.8,
          },
        ]
      }
    >
      FOUR
    </Text>
  </View>
</View>
`;

exports[`renders top TabBar with icons 1`] = `
<View
  style={
    Array [
      Object {
        "borderTopColor": "rgba(33, 33, 33, 0.5)",
        "borderTopWidth": 1,
        "flexDirection": "row",
      },
      Object {
        "backgroundColor": "#486c86",
      },
    ]
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "opacity": 1,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#FFF",
            "fontSize": 20,
          },
          Object {
            "marginRight": 5,
            "opacity": 1,
          },
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "opacity": 1,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "rgba(209, 216, 224, 0.8)",
            "fontSize": 20,
          },
          Object {
            "marginRight": 5,
            "opacity": 0.8,
          },
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "opacity": 1,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "rgba(209, 216, 224, 0.8)",
            "fontSize": 20,
          },
          Object {
            "marginRight": 5,
            "opacity": 0.8,
          },
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "opacity": 1,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "rgba(209, 216, 224, 0.8)",
            "fontSize": 20,
          },
          Object {
            "marginRight": 5,
            "opacity": 0.8,
          },
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`renders top TabBar with tab titles 1`] = `
<View
  style={
    Array [
      Object {
        "borderTopColor": "rgba(33, 33, 33, 0.5)",
        "borderTopWidth": 1,
        "flexDirection": "row",
      },
      Object {
        "backgroundColor": "#486c86",
      },
    ]
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "opacity": 1,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFF",
            "fontSize": 13,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "opacity": 1,
          },
        ]
      }
    >
      ONE
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "opacity": 1,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "rgba(209, 216, 224, 0.8)",
            "fontSize": 13,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "opacity": 0.8,
          },
        ]
      }
    >
      TWO
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "opacity": 1,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "rgba(209, 216, 224, 0.8)",
            "fontSize": 13,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "opacity": 0.8,
          },
        ]
      }
    >
      THREE
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "opacity": 1,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "rgba(209, 216, 224, 0.8)",
            "fontSize": 13,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "opacity": 0.8,
          },
        ]
      }
    >
      FOUR
    </Text>
  </View>
</View>
`;

exports[`renders top TabBar with tab titles and icons 1`] = `
<View
  style={
    Array [
      Object {
        "borderTopColor": "rgba(33, 33, 33, 0.5)",
        "borderTopWidth": 1,
        "flexDirection": "row",
      },
      Object {
        "backgroundColor": "#486c86",
      },
    ]
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "opacity": 1,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#FFF",
            "fontSize": 20,
          },
          Object {
            "marginRight": 5,
            "opacity": 1,
          },
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#FFF",
            "fontSize": 13,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "opacity": 1,
          },
        ]
      }
    >
      ONE
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "opacity": 1,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "rgba(209, 216, 224, 0.8)",
            "fontSize": 20,
          },
          Object {
            "marginRight": 5,
            "opacity": 0.8,
          },
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "rgba(209, 216, 224, 0.8)",
            "fontSize": 13,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "opacity": 0.8,
          },
        ]
      }
    >
      TWO
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "opacity": 1,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "rgba(209, 216, 224, 0.8)",
            "fontSize": 20,
          },
          Object {
            "marginRight": 5,
            "opacity": 0.8,
          },
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "rgba(209, 216, 224, 0.8)",
            "fontSize": 13,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "opacity": 0.8,
          },
        ]
      }
    >
      THREE
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "opacity": 1,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "rgba(209, 216, 224, 0.8)",
            "fontSize": 20,
          },
          Object {
            "marginRight": 5,
            "opacity": 0.8,
          },
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "rgba(209, 216, 224, 0.8)",
            "fontSize": 13,
            "fontWeight": "normal",
            "textAlign": "left",
          },
          Object {
            "opacity": 0.8,
          },
        ]
      }
    >
      FOUR
    </Text>
  </View>
</View>
`;
